src/lib/rules/html-checker/html-checker.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * @fileoverview Validating html using `the Nu html checker`;
+ * https://validator.w3.org/nu/
+ */
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+import { debug as d } from '../../utils/debug';
+import { RuleContext } from '../../rule-context'; // eslint-disable-line no-unused-vars
+import { IRule, IRuleBuilder, ITargetFetchEnd, IScanEnd, IProblemLocation, Severity } from '../../types'; // eslint-disable-line no-unused-vars
+
+const debug: debug.IDebugger = d(__filename);
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The promise that represents the scan by html checker. */
+        let htmlCheckerPromise: Promise<any>;
+        /** Array of strings that needes to be ignored from the checker result. */
+        let ignoredMessages;
+        /** The options to pass to the html checker. */
+        const scanOptions = {
+            data: '',
+            format: 'json',
+            validator: ''
+        };
+
+        type HtmlError = { // eslint-disable-line no-unused-vars
+            extract: string, // code snippet
+            firstColumn: number,
+            lastLine: number,
+            hiliteStart: number,
+            message: string,
+            subType: string
+        };
+
+        const loadRuleConfig = () => {
+            const ignore = (context.ruleOptions && context.ruleOptions.ignore) || [];
+            const validator = (context.ruleOptions && context.ruleOptions.validator) || 'http://validator.w3.org/nu/';
+
+            scanOptions.validator = validator;
+
+            // Up to now, the `ignore` setting in `html-validator` only works if `format` is set to `text`
```

Comment:
If this is a simple change, maybe we should offer and contribute back the fix to it?